plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateCommandWrapper.java

```diff
@@ -149,7 +149,11 @@ Use VIR_DOMAIN_XML_SECURE (value = 1) prior to v1.0.0.
 
             final String target = command.getDestinationIp();
             xmlDesc = dm.getXMLDesc(xmlFlag);
-            xmlDesc = replaceIpForVNCInDescFile(xmlDesc, target);
+
+            // Limit the VNC password in case the length is greater than 8 characters
+            // Since libvirt version 8 VNC passwords are limited to 8 characters
+            String vncPassword = org.apache.commons.lang3.StringUtils.truncate(to.getVncPassword(), 8);
+            xmlDesc = replaceIpForVNCInDescFileAndNormalizePassword(xmlDesc, target, vncPassword);
```

Comment:
separate methos?
```suggestion
            /**
             * Limit the VNC password in case the length is greater than 8 characters
             * Since libvirt version 8 VNC passwords are limited to 8 characters
             */
            private XmlDescOrSuchAThing getXmlDesc(....) {
                String vncPassword = org.apache.commons.lang3.StringUtils.truncate(to.getVncPassword(), 8);
                return replaceIpForVNCInDescFileAndNormalizePassword(xmlDesc, target, vncPassword);
            }
```